src/reporter/index.js

```diff
@@ -100,10 +102,29 @@ export default class Reporter {
             // fixture, we can report this fixture start.
             nextReportItem = this.reportQueue[0];
 
-            await this.plugin.reportTestDone(reportItem.test.name, reportItem.testRunInfo, reportItem.test.meta);
-
-            if (nextReportItem && nextReportItem.fixture !== currentFixture)
-                await this.plugin.reportFixtureStart(nextReportItem.fixture.name, nextReportItem.fixture.path, nextReportItem.fixture.meta);
+            await this.dispatchToPlugin({
```

Comment:
Why do we need to pass enums or strings? Can we just pass and method a execute it with `method.apply(this.plugin, args)`?
I mean to rewrite the `dispatch` function signature to the following:
```JS
dispatchToPlugin(method, ...args)
```

Reply:
I prefer to avoid using `Function.prototype.apply()` whenever I can.
More `OO style`-like `this.plugin[method](...args)` instead of more `functional style`-like `method.apply(this.plugin, args)` gives the natural order of execution, IMO.
Also, passed object with explicit named keys seems to me much more readable than C-like function call
```js
dispatchToPlugin(this.plugin.reportTestDone, reportItem.test.name, reporterItem.testRunInfo, reportItem, test.meta)
```
I want to hear @AndreyBelym's opinion on this.

Reply:
We will still need method names to include them in the error text. So instead of using both method objects and method names, we can just use method names everywhere. Well, we can try to look at `Function.name` to determine it... Likely it should work, but I'm not sure that it will work for all cases.

Due to type checking, the enum is just as safe as referencing the methods directly.

Reply:
Additional +1: if will pass an enum item, we can make `dispatchToPlugin` even more typesafe, by checking the corresponding method args with by looking for the method types by the key and using the [Parameters](https://www.typescriptlang.org/docs/handbook/utility-types.html) generic. It will be really harder (or even not possibile at all) to do if pass a function instead.

Reply:
ok